### ISSUE TITLE
Ignite run lock causes contention on large machines

### ISSUE THREAD

**Issue by eseliger** [sourcegraph/sourcegraph#29524]
We currently need to have a lock around invocations of `ignite run` because it sometimes errors otherwise. This causes contention on machines that want to spin up many VMs though. Maybe we can fix the problem upstream in ignite?

**Comment by chrispine** [comment]
So does this need investigation? It's hard for me to even guess at estimates at the moment.

**Comment by eseliger** [comment]
Ah I actually fixed this -- partly. 
We don't need the runlock at the moment since the _race conditions_ are fixed, but we still see poor performance/timeouts on giant executor machines like the AWS metal machines (which are required on AWS since they are the only ones supporting KVM) due to massive IO contention. While the machine has 72 cores and 192 GB of RAM, the disk it uses is still a regular disk, like on any other server, so it's totally understandable that the IO would suffer heavily if 30 OSes were hammering against it instead of just 1. Still, this is annoying and I want to figure out if there's a way to configure them better. This is most likely going to be a side-project that will get attention every now and then, or when a customer starts complaining about it, but yeah I also struggle hard to put an estimate on this. We might be able to limit IO, but then the contention doesn't go away, it would just prevent timeouts, ideally. Maybe mounting more or more local disks could help, but all I could do is say we time box this to some investigation amount. Due to slow feedback loop, I expect this to be a couple days, excluding the fix, potentially. 
There's likely also room for improvement on the VM creation process, currently we clone the repo first and then copy it, if we were able to control more of the ignite spawn process, we could make that one step and clone the repo into the mounted fs overlay. But that has a prerequisite on getting more control over our VM spawn process.
